arrow-buffer/src/buffer/mutable_ops.rs

```diff
@@ -0,0 +1,1031 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use super::MutableBuffer;
+use crate::bit_chunk_iterator::BitChunks;
+use crate::util::bit_util;
+
+impl MutableBuffer {
+    /// Apply a binary bitwise operation on self (mutate) with respect to another buffer (right).
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - The mutable buffer to be modified in-place
+    /// * `offset_in_bits` - Starting bit offset in Self buffer
+    /// * `right` - slice of bit-packed bytes in LSB order
+    /// * `right_offset_in_bits` - Starting bit offset in the right buffer
+    /// * `len_in_bits` - Number of bits to process
+    /// * `op` - Binary operation to apply (e.g., `|a, b| a & b`)
+    ///
```

Comment:
I think it would help to explain more of how cool this code is

For example:
```suggestion
    /// Apply a binary bitwise operation on self (mutate) with respect to another buffer (right).
    ///
    /// Note: applies the operation a 64-bits (u64) at a time.
    ///
    /// # Arguments
    ///
    /// * `self` - The mutable buffer to be modified in-place
    /// * `offset_in_bits` - Starting bit offset in Self buffer
    /// * `right` - slice of bit-packed bytes in LSB order
    /// * `right_offset_in_bits` - Starting bit offset in the right buffer
    /// * `len_in_bits` - Number of bits to process
    /// * `op` - Binary operation to apply (e.g., `|a, b| a & b`). Applied a word at a time
    ///
```